### PR TITLE
editorconfig-checker 2.5.0 (new formula)

### PR DESCRIPTION
**Pull request by sudosubin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/editorconfig-checker.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class EditorconfigChecker < Formula
+  desc "Tool to verify that your files are in harmony with your .editorconfig"
+  homepage "https://github.com/editorconfig-checker/editorconfig-checker"
+  url "https://github.com/editorconfig-checker/editorconfig-checker/archive/refs/tags/#{version}.tar.gz"
+  version "2.5.0"
```
Comment (carlocab) [inline Formula/editorconfig-checker.rb:5]:
```suggestion
  url "https://github.com/editorconfig-checker/editorconfig-checker/archive/refs/tags/2.5.0.tar.gz"
```

File: Formula/editorconfig-checker.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  sha256 "50274a09c760303bc88c0ead319eda4bc6f317764a1cade1ea86e9ff17d7c0b1"
+  license "MIT"
+  head "https://github.com/editorconfig-checker/editorconfig-checker.git", branch: "main"
+
+  depends_on "go@1.16" => :build
```
Comment (carlocab) [inline Formula/editorconfig-checker.rb:10]: Can we not use the latest Go?

File: Formula/editorconfig-checker.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-X main.version=#{version}", output: bin/"editorconfig-checker"), "./cmd/editorconfig-checker/main.go"
```
Comment (carlocab) [inline Formula/editorconfig-checker.rb:13]:
```suggestion
    system "go", "build", *std_go_args(ldflags: "-X main.version=#{version}"), "./cmd/editorconfig-checker/main.go"
```
You don't need this.

File: Formula/editorconfig-checker.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/editorconfig-checker --version")
```
Comment (carlocab) [inline Formula/editorconfig-checker.rb:17]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Review by carlocab — COMMENTED** [review]
```
❯ brew audit --new editorconfig-checker
editorconfig-checker:
  * Stable: The source URL https://github.com/editorconfig-checker/editorconfig-checker/archive/refs/tags/.tar.gz is not reachable (HTTP status code 400)
```
Please don't mark items in the PR template as completed if they're not done yet.

File: Formula/editorconfig-checker.rb (hunk 4 above)
Comment (sudosubin) [reply Formula/editorconfig-checker.rb:21]: thanks, I added a simple test to check a file.

File: Formula/editorconfig-checker.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  def install
+    system "go", "build", *std_go_args(ldflags: "-X main.version=#{version}"), "./cmd/editorconfig-checker/main.go"
+  end
+
+  test do
+    (testpath/"version.txt").write <<~EOS
+      version=#{version}
+    EOS
+
+    system "#{bin}/editorconfig-checker", testpath/"version.txt"
```
Comment (carlocab) [inline Formula/editorconfig-checker.rb:20]:
```suggestion
    system bin/"editorconfig-checker", testpath/"version.txt"
```

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request closed without merging**